cmd/tunneld/tunneld.go

```diff
@@ -95,7 +105,9 @@ func main() {
 			fatal("failed to start HTTPS: %s", s.ListenAndServeTLS(opts.tlsCrt, opts.tlsKey))
 		}()
 	}
-
+	if opts.letsEncrypt {
```

Comment:
I suggest moving this to L76 and putting normal start in else branch so that the additional checks can be avoided.